### PR TITLE
Add Spanish support

### PR DESCRIPTION
**Pull request by jerivas**
- [x] Add spanish constants
- [ ] Add RVG Bible
- [ ] Add BES Bible
- [ ] Add FTL files
- [ ] Translate CMS content
- [ ] Anything else @spookylukey?

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #
 #    pip-compile --output-file=requirements.txt requirements-dev.in requirements.in
 #
--e git+git://github.com/riccardoscalco/Pykov.git@16cda2f2c0ed8143ddcfa25dac34d3dadc982380#egg=pykov
+pykov @ https://github.com/riccardoscalco/Pykov/archive/16cda2f2c0ed8143ddcfa25dac34d3dadc982380.tar.gz
```
Comment (jerivas) [inline requirements.txt:7]: Unrelated, but installing from a tarball is faster than cloning the entire repo

File: package.json [hunk 2]
```diff
@@ -38,7 +38,7 @@
   "dependencies": {
     "autosize": "^4.0.0",
     "cal-heatmap": "^3.6.2",
-    "elm-webpack-loader": "git://github.com/CareerPlug/elm-webpack-loader.git#4.5.1",
+    "elm-webpack-loader": "CareerPlug/elm-webpack-loader#4.5.1",
```
Comment (jerivas) [inline package.json:41]: Had some trouble with the complete git syntax, but this works fine

**Review by jerivas — COMMENTED** [review]
I snuck in some requirement cleanup, I can open a separate PR if preferred

**Comment by spookylukey** [comment]
This looks great, the extra cleanups are fine.

For the Spanish constants, ideally we'd also have an addition to `make_bible_book_abbreviations` which involves creating `ES_EXTRA_BOOK_NAMES`  after the pattern of `EN_EXTRA_BOOK_NAMES`.  This helps people when entering references, otherwise they need the full book name like "1 Tesalonicenses" or "El Cantar de los Cantares" etc.

**Comment by spookylukey** [comment]
Also, for Bible references. The reference parser currently assumes:

    # Bible references look something like:
    # Genesis
    # Genesis 1
    # Genesis 1:2
    # Genesis 1:2-3
    # Genesis 1:2-4:5


If there are other alternatives in common in Spanish that you think we should support, please let me know.

**Comment by spookylukey** [comment]
If you are happy with this, my suggestion is that I go ahead and deploy the RVG, and then do translation of the site as a separate step.

**Comment by spookylukey** [comment]
Also, I'll put you as a "tester" on learnscripture.net so that you can test the RVG before making it public.

**Comment by spookylukey** [comment]
I've added RVG online now, you should be able to see it and use it etc. It would be great if you could give it a thorough try out, including setting as your default bible, full text search of verses etc. Then we can switch it to public, and work on other site localization

**Comment by jerivas** [comment]
Cool! I just tested the following:

- Setting RVG as the default Bible, works ok
- Went through the Bible 101 set, all verses look good
- Passage search (including abbreviations), all good
- Full text search. Performed several searches and got the results I expected, including the correct handling of accented characters

The only thing that is missing is the "Choose from word list" learning strategy. When I get to that part in the memorization I get:

```
On screen testing is not available for this verse in this version. Sorry! Please choose another option in your preferences
```

**Pull request merged**